readthedocs/oauth/services/gitlab.py

```diff
@@ -55,7 +55,7 @@ def _get_repo_id(self, project):
         # The ID or URL-encoded path of the project
         # https://docs.gitlab.com/ce/api/README.html#namespaced-path-encoding
         try:
-            repo_id = json.loads(project.remote_repository.json).get('id')
+            repo_id = project.remote_repository.remote_id
         except Exception:
```

Comment:
We could probably remove the try/except clause here as well.

Reply:
if the project was imported manually it would throw an exception as stated here.

https://github.com/readthedocs/readthedocs.org/blob/349ba5a3f83efacb3b7a5c1005d81dc3d7332185/readthedocs/oauth/services/gitlab.py#L60

Reply:
He, I didn't read the following line.

However, the previous case would have failed because the `json.loads` would have failed. Now, accessing `remote_id` will return just `None` and won't fail. I think we should check `if project.remote_repository.remote_id is None` instead of the try/except, right?

Reply:
In this case the `project.remote_repository` wont exist and throw `RemoteRepositroy.DoesNotExist` Exception.

Reply:
Good point! Can we catch that exception here then instead of the generic `Exception`?